slides/k8s/build-with-kaniko.md

```diff
@@ -0,0 +1,209 @@
+# Building images with Kaniko
+
+- [Kaniko](https://github.com/GoogleContainerTools/kaniko) is an open source tool to build container images within Kubernetes
+
+- It can build an image using any standard Dockerfile
+
+- The resulting image can be pushed to a registry or exported as a tarball
+
+- It doesn't require any particular privilege
+
+  (and can therefore run in a regular container in a regular pod)
+
+- This combination of features is pretty unique
+
+  (most other tools use different formats, or require elevated privileges)
+
+---
+
+## Kaniko in practice
+
+- Kaniko provides an "executor image", `gcr.io/kaniko-project/executor`
+
+- When running that image, we need to specify at least:
+
+  - the path to the build context (=the directory with our Dockerfile)
+
+  - the target image name (including the registry address)
+
+- Simplified example:
+  ```
+  docker run \
+      -v ...:/workspace gcr.io/kaniko-project/executor \
+      --context=/workspace \
+      --destination=registry:5000/image_name:image_tag
+  ```
+
+---
+
+## Running Kaniko in a Docker container
+
+- Let's build the image for the DockerCoins `worker` service with Kaniko
+
+.exercise[
+
+- Find the port number for our self-hosted registry:
+  ```bash
+  kubectl get svc registry
+  PORT=$(kubectl get svc registry -o json | jq .spec.ports[0].nodePort)
+  ```
+
+- Run Kaniko:
+  ```bash
+  docker run --net host \
+      -v ~/container.training/dockercoins/worker:/workspace \
+      gcr.io/kaniko-project/executor \
+      --context=/workspace \
+      --destination=127.0.0.1:30448/worker-kaniko:latest 
+  ```
+
+]
+
+We use `--net host` so that we can connect to the registry over `127.0.0.1`.
+
+---
+
+## Running Kaniko in a Kubernetes pod
+
+- We need to mount or copy the build context to the pod
+
+- For bonus points, we are going to build straight from the git repository
```

Comment:
"bonus points" is funny, but as someone new to Kaniko, it's unclear _why_ this is... better? is that what "bonus points" is conveying here?

Reply:
Right! I wanted to convey the idea that this was better because the workflow would happen entirely in containers (without needing files sitting on a node, outside of containers).

Well then: 

- We are going to build straight from the git repository

  (to avoid depending on files sitting on a node, outside of containers)